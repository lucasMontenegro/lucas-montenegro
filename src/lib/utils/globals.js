import g from "global"
// renaming "global" to "globals"
// declaring a variable named "global" may cause trouble
export default g