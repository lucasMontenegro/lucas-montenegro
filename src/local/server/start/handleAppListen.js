const { process, console } = global
module.exports = function handleAppListen (PORT) {
  return () => {
    const name = (
      process.env.NODE_ENV === `production` ? `cluster worker ${process.pid}` : `dev server`
    )
    console.log(`Node ${name}: listening on port ${PORT}`)
  }
}