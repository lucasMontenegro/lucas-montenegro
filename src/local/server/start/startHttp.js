const express = require("express")
const api = require("local/server/start/api")
const sendFrontEndApp = require("local/server/start/sendFrontEndApp")
const handleAppListen = require("local/server/start/handleAppListen")
const { process } = global
module.exports = function startHttp () {
  const app = express()
  // Priority serve any static files.
  app.use(express.static(process.env.BUILD_PATH))
  // Answer API requests.
  app.use(`/api`, api())
  // All remaining requests return the React app, so it can handle routing.
  app.get(`*`, sendFrontEndApp)
  const PORT = process.env.PORT || 5000
  app.listen(PORT, handleAppListen(PORT))
}