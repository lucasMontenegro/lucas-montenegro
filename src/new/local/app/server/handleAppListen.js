const { default: globals } = require("new/local/utils/globals")
const { default: isProduction } = require("new/local/utils/isProduction")
exports.default = function handleAppListen (PORT) {
  return () => {
    const name = isProduction() ? `cluster worker ${globals.process.pid}` : `dev server`
    globals.console.log(`Node ${name}: listening on port ${PORT}`)
  }
}