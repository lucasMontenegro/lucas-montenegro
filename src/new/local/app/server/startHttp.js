const express = require("express")
const { default: api } = require("new/local/app/server/api")
const { default: sendFrontEndApp } = require("new/local/app/server/sendFrontEndApp")
const { default: handleAppListen } = require("new/local/app/server/handleAppListen")
const { default: globals } = require("new/local/utils/globals")
exports.default = function startHttp () {
  const app = express()
  // Priority serve any static files.
  app.use(express.static(globals.process.env.BUILD_PATH))
  // Answer API requests.
  app.use(`/api`, api)
  // All remaining requests return the React app, so it can handle routing.
  app.get(`*`, sendFrontEndApp)
  const PORT = globals.process.env.PORT || 5000
  app.listen(PORT, handleAppListen(PORT))
}