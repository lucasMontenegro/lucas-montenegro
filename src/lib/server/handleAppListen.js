import globals from "lib/utils/globals"
import isProduction from "lib/utils/isProduction"
export default function handleAppListen (PORT) {
  return () => {
    const name = isProduction() ? `cluster worker ${globals.process.pid}` : `dev server`
    globals.console.log(`Node ${name}: listening on port ${PORT}`)
  }
}