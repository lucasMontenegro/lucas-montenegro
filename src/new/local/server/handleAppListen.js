import globals from "new/local/utils/globals"
import isProduction from "new/local/utils/isProduction"
export default function handleAppListen (PORT) {
  return () => {
    const name = isProduction() ? `cluster worker ${globals.process.pid}` : `dev server`
    globals.console.log(`Node ${name}: listening on port ${PORT}`)
  }
}