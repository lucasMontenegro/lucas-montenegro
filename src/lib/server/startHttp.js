import express from "express"
import sendFrontEndApp from "lib/server/sendFrontEndApp"
import handleAppListen from "lib/server/handleAppListen"
import globals from "lib/utils/globals"
export default function startHttp (api) {
  const app = express()
  // Priority serve any static files.
  app.use(express.static(globals.process.env.BUILD_PATH))
  // Answer API requests.
  app.use(`/api`, api)
  // All remaining requests return the React app, so it can handle routing.
  app.get(`*`, sendFrontEndApp)
  const PORT = globals.process.env.PORT || 5000
  app.listen(PORT, handleAppListen(PORT))
}