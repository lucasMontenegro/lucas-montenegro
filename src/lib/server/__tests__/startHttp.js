import express from "express"
import handleAppListen from "lib/server/handleAppListen"
import globals from "lib/utils/globals"
import startHttp from "lib/server/startHttp"
const mockFn = jest.fn()
const makeFn = name => (...args) => {
  const arr = [name, args]
  mockFn(...arr)
  return arr
}
jest.mock(`express`, () => ({ __esModule: true, default: jest.fn() }))
express.mockImplementation((...args) => {
  mockFn(`express`, args)
  return [`use`, `get`, `listen`].reduce((app, name) => {
    app[name] = makeFn(`app.${name}`)
    return app
  }, {})
})
express.static = makeFn(`express.static`)
jest.mock(`lib/server/sendFrontEndApp`, () => ({
  __esModule: true,
  default: `lib/server/sendFrontEndApp`,
}))
jest.mock(`lib/server/handleAppListen`, () => ({
  __esModule: true,
  default: jest.fn(),
}))
handleAppListen.mockImplementation(makeFn(`handleAppListen`))
jest.mock(`lib/utils/globals`, () => ({
  __esModule: true,
  default: {
    process: { env: { BUILD_PATH: `'process.env.BUILD_PATH'` } },
  },
}))
describe(`lib/server/startHttp`, () => {
  afterEach(() => mockFn.mockClear())
  it(`should use the right dependency versions`, () => {
    expect(jestUtils.getDependencies([`express`])).toMatchSnapshot()
  })
  test.each([
    [80],
    [null],
  ])(`should configure and start the http app (PORT %j)`, PORT => {
    globals.process.env.PORT = PORT
    startHttp(`api`)
    expect(mockFn.mock.calls).toMatchSnapshot()
  })
})