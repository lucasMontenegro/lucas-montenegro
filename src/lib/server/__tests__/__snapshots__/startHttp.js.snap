// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lib/server/startHttp should configure and start the http app (PORT 80) 1`] = `
Array [
  Array [
    "express",
    Array [],
  ],
  Array [
    "express.static",
    Array [
      "'process.env.BUILD_PATH'",
    ],
  ],
  Array [
    "app.use",
    Array [
      Array [
        "express.static",
        Array [
          "'process.env.BUILD_PATH'",
        ],
      ],
    ],
  ],
  Array [
    "app.use",
    Array [
      "/api",
      "api",
    ],
  ],
  Array [
    "app.get",
    Array [
      "*",
      "lib/server/sendFrontEndApp",
    ],
  ],
  Array [
    "handleAppListen",
    Array [
      80,
    ],
  ],
  Array [
    "app.listen",
    Array [
      80,
      Array [
        "handleAppListen",
        Array [
          80,
        ],
      ],
    ],
  ],
]
`;

exports[`lib/server/startHttp should configure and start the http app (PORT null) 1`] = `
Array [
  Array [
    "express",
    Array [],
  ],
  Array [
    "express.static",
    Array [
      "'process.env.BUILD_PATH'",
    ],
  ],
  Array [
    "app.use",
    Array [
      Array [
        "express.static",
        Array [
          "'process.env.BUILD_PATH'",
        ],
      ],
    ],
  ],
  Array [
    "app.use",
    Array [
      "/api",
      "api",
    ],
  ],
  Array [
    "app.get",
    Array [
      "*",
      "lib/server/sendFrontEndApp",
    ],
  ],
  Array [
    "handleAppListen",
    Array [
      5000,
    ],
  ],
  Array [
    "app.listen",
    Array [
      5000,
      Array [
        "handleAppListen",
        Array [
          5000,
        ],
      ],
    ],
  ],
]
`;

exports[`lib/server/startHttp should use the right dependency versions 1`] = `
Object {
  "express": "^4.17.1",
}
`;
